kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepository.java

```diff
@@ -0,0 +1,151 @@
+                         .stream()
+                         .map(cfg -> cfg.getMetadata().getName())
+                         .collect(Collectors.toList());
+        });
+    }
+
+    public List<KieServerState> retrieveAllKieServerStates() {
+        return processKieServerStateByOpenShift(client -> {
+            return client.configMaps().withLabel(CFG_MAP_LABEL_NAME, CFG_MAP_LABEL_VALUE).list().getItems()
+                         .stream()
+                         .map(cfg -> (KieServerState) xs.fromXML(cfg.getData().get(CFG_MAP_DATA_KEY)))
+                         .collect(Collectors.toList());
+        });
+    }
+
+    public boolean exists(String id) {
+        return processKieServerStateByOpenShift(client -> client.configMaps().withName(id).get() != null);
+    }
+
+    public KieServerState delete(String id) {
+        KieServerState state = load(id);
+        boolean isUnsupported = processKieServerStateByOpenShift(client -> {
+            if (client.deploymentConfigs().withName(id).get() == null) {
+                client.configMaps().withName(id).delete();
+            } else {
+                return true;
+            }
+            return false;
+        });
+
+        if (isUnsupported) {
+            logger.error("Can not delete attached KieServerState with id [{}].", id);
+            throw new UnsupportedOperationException("Can not delete attached KieServerState with id [" + id + "]");
+        }
+        return state;
+    }
+
+    @Override
+    public synchronized void store(@NotNull String serverId, @NotNull KieServerState kieServerState) {
+        if (!retrieveKieServerId(kieServerState).equals(serverId)) {
+            throw new IllegalArgumentException("Invalid KieServerId: Id does not match with KieServerState.");
+        }
+
+        processKieServerStateByOpenShift(client -> {
+            DeploymentConfig dc = client.deploymentConfigs().withName(serverId).get();
+            ConfigMap cm = client.configMaps().withName(serverId).get();
+            String stateXML = xs.toXML(kieServerState);
+            if (cm == null) {
+                throw new IllegalStateException("KieServerState ConfigMap must exist before update.");
+            } else {
+                cm.setData(Collections.singletonMap(CFG_MAP_DATA_KEY, stateXML));
+            }
+
+            ObjectMeta md = cm.getMetadata();
+            Map<String, String> ann = md.getAnnotations() == null ? new ConcurrentHashMap<>() : md.getAnnotations();
+            md.setAnnotations(ann);
+            ann.put(STATE_CHANGE_TIMESTAMP,
+                    ZonedDateTime.now().format(DateTimeFormatter.ISO_INSTANT));
+
+            if (isKieServerReady()) {
+                if (isDCStable(dc)) {
+                    ann.put(ROLLOUT_REQUIRED, "true");
+                    client.configMaps().createOrReplace(cm);
+                } else {
+                    throw new UnsupportedOperationException(
```

Comment:
The exception is not propagated back to the KieServerImpl. It is caught by https://github.com/kiegroup/droolsjbpm-integration/pull/1611/files#diff-92456e3292062ce4cd2fdeec049e52fdR147 . So Kie server will return success message (HTTP 201) and container will not be stored.